lib/dev-requirements.txt

```diff
@@ -1,4 +1,4 @@
-black>=22.3.0
+black==22.6.0
```

Comment:
What is the reason to pin this?

Reply:
In pre-commit the version is pinned at 22.6.0. So if you run 'make pylint' it'll run black's latest version (>= 22.3.0), and then the files fail the pre-commit check when you try to commit them because the two versions have slightly different rules.

Maybe 'make pylint' is meant to be deprecated in favor of 'make pyformat' which runs the _pre-commit_ version of black anyways...?

Reply:
@LukasMasuch FYI the constraints for 3.12 currently break with this black pin.
![image](https://github.com/streamlit/streamlit/assets/5677417/c7abe520-cd16-4fe5-8fef-3dc2b0a8db40)
I'm not really sure what the opinionatedly correct way to deal with this is...maybe just remove black from the dev dependencies and change `make pylint` to print a deprecation message? or I can submit a separate PR to the constraints-3.12 branch?

Reply:
I believe this will resolve automatically, I think we can just merge as it is.